File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.java
class Solution {
    public int solution(int balls, int share) {
        return combination(balls,share);
    }
    public int combination(int n,int r){
        if(r>n-r){
            r=n-r;
        }
        long result=1;
        
        for(int i=0;i<r;i++){
            result*=(n-i);
            result/=(i+1);
        }
        return (int)result;
    }
}

File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.js
function solution(balls, share) {
    let sum1=1;
    let sum2=1;
    for (let x=balls;x>=balls-share+1;x--){
      
        sum1*=x;
    }
    for(let y=1;y<=share;y++){
       
        sum2*=y;
    }
    return Math.round(sum1/sum2);
}